File: daily/Linked List/2130-Maximum-twin-sum-of-a-linked-list.py
class Solution:
    def pairSum(self, head: Optional[ListNode]) -> int:
        slow = fast = head
        while fast and fast.next:
            slow = slow.next
            fast = fast.next.next
        
        prev = None
        curr = slow
        while curr:
            next_node = curr.next
            curr.next = prev
            prev = curr
            curr = next_node
        
        maxTwinSum = 0
        left, right = head, prev
        while right:
            currentTwinSum = left.val + right.val
            maxTwinSum = max(maxTwinSum, currentTwinSum)
            left = left.next
            right = right.next

        return maxTwinSum

File: daily/Linked List/2130-Maximum-twin-sum-of-a-linked-list.go
package daily

func pairSum(head *ListNode) int {
	slow := head
	fast := head
	for fast != nil && fast.Next != nil {
		slow = slow.Next
		fast = fast.Next.Next
	}

	var prev *ListNode
	curr := slow
	for curr != nil {
		next := curr.Next
		curr.Next = prev
		prev = curr
		curr = next
	}

	maxTwinSum := 0
	left := head
	right := prev
	for right != nil {
		currentTwinSum := left.Val + right.Val
		if currentTwinSum > maxTwinSum {
			maxTwinSum = currentTwinSum
		}
		left = left.Next
		right = right.Next
	}

	return maxTwinSum
}
